DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/lacp/test_lacp_routing_over_vrf.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+from math import isclose
+import pytest
+
+
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.lib.ip.ip_address import IpAddress
+from dent_os_testbed.lib.ip.ip_route import IpRoute
+
+
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_traffic_generator_connect,
+    tgen_utils_dev_groups_from_config,
+    tgen_utils_get_traffic_stats,
+    tgen_utils_setup_streams,
+    tgen_utils_start_traffic,
+    tgen_utils_stop_traffic,
+    tgen_utils_get_swp_info,
+)
+
+pytestmark = [pytest.mark.suite_funtional_lacp,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures('cleanup_tgen', 'cleanup_bonds',
+                                      'cleanup_bridges', 'cleanup_sysctl',
+                                      'cleanup_vrfs', 'cleanup_routes',
+                                      'cleanup_tables', 'enable_ipv4_forwarding')]
+
+
+async def test_lacp_routing_over_lag_vrf(testbed):
+    """
+    Test Name: LACP routing
+    Test Suite: suite_functional_lacp
+    Test Overview: Test routing over vrf's
+    Test Procedure:
+    1. Create 3 bonds and 2 vrf tables
+    2. Enslave
+        DUT port <==> tgen port 1 to bond 1
+        DUT port <==>  tgen port 2 to bond 2
+        DUT port <==> tgen port 3 to bond 3
+        DUT port <==>  tgen port 4 to bond 3
+    3. Set link up on all participant ports
+    4. Configure ip addresses in each LAG and configure route 10.1.1.0/24 via 2 vrf's
+    5. Setup one stream with destination IP 10.1.1.1
+    6. Transmit traffic
+    7. Verify traffic received on active bond2 port in vrf0 and not received on vrf1
+    """
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        pytest.skip('The testbed does not have enough dent with tgen connections')
+    device = dent_devices[0]
+    dent = device.host_name
+    tg_ports = tgen_dev.links_dict[dent][0]
+    ports = tgen_dev.links_dict[dent][1]
+    bonds = [f'bond_{idx}' for idx in range(1, 4)]
+    tgen_lag_1 = ('LAG_0', [tg_ports[0]])
+    tgen_lag_2 = ('LAG_1', [tg_ports[1]])
+    tgen_lag_3 = ('LAG_2', tg_ports[2:])
```

Comment:
suggestion: you can use the same name for the lags on ixia as the name for bonds on dut (bond_1, bond_2, ...)